modules/trino/src/main/java/org/testcontainers/containers/TrinoContainer.java

```diff
@@ -0,0 +1,109 @@
+package org.testcontainers.containers;
+
+import org.jetbrains.annotations.NotNull;
+import org.testcontainers.containers.wait.strategy.LogMessageWaitStrategy;
+import org.testcontainers.utility.DockerImageName;
+
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.time.Duration;
+import java.util.HashSet;
+import java.util.Set;
+
+import static com.google.common.base.Strings.nullToEmpty;
+import static java.lang.String.format;
+import static java.time.temporal.ChronoUnit.SECONDS;
+
+/**
+ * @author findepi
+ */
+public class TrinoContainer<SELF extends TrinoContainer<SELF>> extends JdbcDatabaseContainer<SELF> {
+    public static final String NAME = "trino";
+    private static final DockerImageName DEFAULT_IMAGE_NAME = DockerImageName.parse("trinodb/trino");
+    public static final String IMAGE = "trinodb/trino";
+    public static final String DEFAULT_TAG = "351";
+
+    public static final Integer TRINO_PORT = 8080;
+
+    private String username = "test";
+    private String catalog = null;
+
+    public TrinoContainer(final String dockerImageName) {
+        this(DockerImageName.parse(dockerImageName));
+    }
+
+    public TrinoContainer(final DockerImageName dockerImageName) {
+        super(dockerImageName);
+
+        dockerImageName.assertCompatibleWith(DEFAULT_IMAGE_NAME);
+
+        this.waitStrategy = new LogMessageWaitStrategy()
```

Comment:
as a JDBC-style container, the wait strategy will use the driver and perform a test query, so perhaps we should drop this log-based strategy

Reply:
This is the convention we have in the project that the `SERVER STARTED` should be considered as 'the official'.

Reply:
The thing is... `JdbcDatabaseContainer` overrides `waitUntilContainerStarted` and uses `SELECT 1`-based approach. So this line becomes a no-op

Reply:
removed